File: ToyLanguageLock/src/Model/MyStack.java
package Model;

import com.sun.deploy.util.ArrayUtil;

import java.util.ArrayList;
import java.util.Collection;
import java.util.Collections;
import java.util.EmptyStackException;

public class MyStack<T> implements MyIStack<T> {
    private ArrayList<T> stack;

    public MyStack() {
        this.stack = new ArrayList<T>();
    }

    @Override
    public T pop() {
        if(!isEmpty())
        {
            return stack.remove(size()-1);
        }
        else
            throw new EmptyStackException();

    }

    @Override
    public void push(T v) {
        stack.add(v);
    }

    @Override
    public boolean isEmpty() {
        return (stack.size() == 0);
    }

    @Override
    public int size() {
        return stack.size();
    }

    @Override
    public ArrayList<String> getAllStack() {
        ArrayList<String> arr = new ArrayList<>();
        stack.forEach(el->arr.add(el.toString()));
        Collections.reverse(arr);
        return arr;
    }

    @Override
    public T top() {
        if(!isEmpty())
            return stack.get(stack.size()-1);
        else
            throw new EmptyStackException();
    }

    @Override
    public String toString() {
        return String.valueOf(stack);
    }
}

File: ToyLanguage7/ToyLanguage7/MyStack.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace ToyLanguage7
{
    class MyStack<T> : MyIStack<T>
    {
        private List<T> stack;

        public MyStack() { this.stack = new List<T>(); }

        public bool isEmpty()
        {
            return stack.Count()==0;
        }

        public T pop()
        {
            if (!isEmpty())
            {
                T getTop = this.stack[size() - 1];
                stack.RemoveAt(size() - 1);
                return getTop;
            }
            else
                throw new MyException("Empty stack!");
        }

        public void push(T v)
        {
            stack.Add(v);
        }

        public int size()
        {
            return stack.Count();
        }

        public override string ToString()
        {
            string result = "[";
            List<T> newStack = this.stack;
            newStack.Reverse();
            foreach (var el in newStack)
                result += el + " ";
            result += "]";
            //result = result.Remove(result.Length - 1, 1);
            return result;
        }
    }
}
